src/NUnitEngine/nunit.engine.tests/Drivers/NUnit3FrameworkDriverTests.cs

```diff
@@ -35,6 +35,7 @@ namespace NUnit.Engine.Drivers.Tests
     public class NUnit3FrameworkDriverTests
     {
         private const string MOCK_ASSEMBLY = "mock-assembly.dll";
+        private const string NUNIT_FRAMEWORK_NAME = "nunit.framework, Version=3.9.0.0, Culture=neutral, PublicKeyToken=2638cd05610744eb";
```

Comment:
We have no way of knowing that the version of the framework used. That is, not unless we require installation of an entirely new engine for each framework version.

Reply:
This is the version used by mock-assembly which is part of the same project and which has the version number hardcoded into its csproj:

    <Reference Include="nunit.framework, Version=3.9.0.0, Culture=neutral, PublicKeyToken=2638cd05610744eb, processorArchitecture=MSIL">
      <HintPath>..\..\..\packages\NUnit.3.9.0\lib\net20\nunit.framework.dll</HintPath>
    </Reference>

Reply:
The test assemblies naturally all reference a particular version of the NUnit framework. That's the way NUnit is used and in our tests we are users of NUnit. We usually update the engine tests to the latest release of the framework when we do a new engine release, but there's no requirement to do so unless we are using new features.

There are, however, no references to the framework from the code that we package. Users may use any version of the framework starting with 3.0. This has always been a design goal of NUnit since it's beginning and NUnit 3 continued it. Many users have different projects using different NUnit framework versions but all running under the latest release of some runner.

I think you will do better if you go back to the issue and help us confirm that there is a problem.

Reply:
>There are, however, no references to the framework from the code that we package. Users may use any version of the framework starting with 3.0.

That's not relevant here since my reference is specifically to the nunit.framework used in mock-assembly which can most certainly not be "any version", since it's hard coded in the csproj.

Reply:
This would still be good to avoid, however, since it gives us an extra place to change the framework version when we update it, rather than just changing the reference.

Reply:
This could be fixed by pulling the assembly name out of mock-assembly. Code can be brief since it's guaranteed to have the reference.

Reply:
Let's make this `static readonly` and pull the reference from the current assembly, since it's easy.

Reply:
Also, HintPath doesn't meet my definition of 'hardcoding', nor do NuGet-restore-generated references. What's really hardcoded from my point of view is the NuGet package version 3.9.0.